addons/purchase/models/account_invoice.py

```diff
@@ -194,6 +220,14 @@ def _anglo_saxon_purchase_move_lines(self, i_line, res):
                                 for child in tax.children_tax_ids:
                                     if child.type_tax_use != 'none':
                                         tax_ids.append((4, child.id, None))
+
+                        if inv.currency_id.id != company_currency.id:
+                            valuation_price_unit = (
```

Comment:
So the same variable name now changes currency... not very clear. Maybe use a new variable:
`valuation_pu_inv_curr`